src/homelogstore/log_store.cpp

```diff
@@ -352,7 +356,8 @@ nlohmann::json HomeLogStore::dump_log_store(const log_dump_req& dump_req) {
             }
             json_records.emplace_back(std::move(json_val));
             decltype(idx) end_idx{std::min(max_idx, dump_req.end_seq_num)};
-            end_iterate = (cur_idx < end_idx) ? true : false;
+            end_iterate = (cur_idx < end_idx && --batch_size > 0) ? true : false;
```

Comment:
shouldn't it be --batch_size == 0 ?

Reply:
I modified the variable name to proceed to make it clear.